### ISSUE TITLE
RDF/CDF matrices should call scipy for rank

### ISSUE THREAD

**Issue by jasongrout** [sagemath/sage#7392]
<div id="comment:0"></div>

Right now, we rely on the horrible (for floating point) generic echelon implementation to calculate the rank of an RDF/CDF matrix.

In fact, over RR, it wouldn't hurt to call numpy/scipy either.

Component: **linear algebra**

_Issue created by migration from https://trac.sagemath.org/ticket/7392_

**Comment by videlec** [comment]
I would suggest the straightforward
```
sage: import numpy as np
sage: m=matrix(2, [1.5,1.75,1.5,1.75])
sage: np.linalg.matrix_rank(m.numpy())
1
```

**Comment by HenryWu2101** [comment]
It seems the problem is still unresolved. The rank of "m=matrix(2, [1.5,1.75,1.5,1.75])" is 2 equiv. "m=matrix(RR, [1.5,1.75,1.5,1.75])"; while the rank of "m=matrix(RDF, [1.5,1.75,1.5,1.75])" is 1, which would be correct in this case. The matrix over RR rank method is implemented with unreliable echelon method with poor performance coming to floating point issues. Is the direct "calling numpy/scipy" method still desirable at the moment? The numpy.linalg.matrix_rank() would perform well with the minor issue that it only works with double precision per numpy requirements as I recall.  @tscrim Is it ok if you could also take a look?

**Comment by tscrim** [comment]
I don't know much about these implementations, so I cannot contribute anything useful to the discussion.

**Comment by HenryWu2101** [comment]
Of course, that makes sense. Is there anyone specific whom I should @ or cc to request a review of my PR?
